### DISCUSSION TITLE
Question: is it possible to automate trace in dotnet-monitor and upload to S3

### DISCUSSION THREAD

**Discussion by jaya-rama** [dotnet/dotnet-monitor#5636] [Q&A] [answered]
I'm using dotnet 6 application running in Kubernetes and I have to run dotnet-monitor as a side car and automate trace creation if cpu goes beyond x% and upload it to S3 is this possible with dotnet-monitor if so can you help me with steps and the image that I should use

**Comment by jander-msft** [comment, marked as answer]
This is an existing example of how to collect a trace based on cpu usage: https://github.com/dotnet/dotnet-monitor/blob/main/documentation/collectionrules/collectionruleexamples.md#collect-trace---high-cpu-usage-eventcounter-trigger. With that, you just need to adjust the cpu threshold that you need and configure the egress provider. Here's the docs for the S3 egress provider: https://github.com/dotnet/dotnet-monitor/blob/main/documentation/configuration/egress-configuration.md#80-s3-storage-egress-provider
